### ISSUE TITLE
KeyError while creating new expectation suite for postgres database

### ISSUE THREAD

**Issue by ghost** [fivetran/great_expectations#2936]
**Describe the bug**
While trying to create a new v3 expectations suite with "automatically, using a profiler" a KeyError occues.
As a datasource iam using a postgres database. 

The keyError exactly occues after i provided the schema name, table and suite name.

My schema name: public
My table name: branches

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/great_expectations/datasource/data_connector/configured_asset_sql_data_connector.py", line 150, in get_batch_definition_list_from_batch_request
    sub_cache = self._data_references_cache[batch_request.data_asset_name]
KeyError: 'public.branches'
.....
KeyError: 'data_asset_name public.branches is not recognized.'

**To Reproduce**
1. Create a postgres datasource
2. create a new expectation suite with "great_expectations --v3-api suite new"
3. Automatically, using a profiler
4. Enter a table name and schema
5. Please provide the schema name of the table (this is optional) [public]:
6. Please provide the table name (this is required): branches
7. Name the new Expectation Suite [public.branches.warning]: 
8. Would you like to proceed? [Y/n]: 
9. See error

**Expected behavior**
Created a Expectation Suite. With great expectation v2 I could create a new suite without any problems

**Environment (please complete the following information):**
 - MacOS
 - Great Expectations Version: [e.g. 0.13.19]

**Comment by talagluck** [comment]
Hi @rikardomarenzzi - thanks for your question, and apologies for delays! This appears to be a bug, however there is a workaround. You can specify the `schema_name` in a `batch_spec_passthrough` dictionary in your BatchRequest like so:
```
batch_request = {
    "datasource_name": "my_postgres_datasource",
    "data_connector_name": "default_inferred_data_connector_name",
    "data_asset_name": "my_table_name",
    "limit": 1000,
    "batch_spec_passthrough":{
        "schema_name": "my_schema_name"
    }
}
```

**Comment by mpalfi** [comment]
I opened 3649 a few weeks ago and I was redirected here to try the workaround for this bug.
I tried what it was suggested but ran into a new error. Can you please check and let me know your findings?

Thank you:

The request is like below:
batch_request = {
    "datasource_name": "bdl_dev_v3",
    "data_connector_name": "default_inferred_data_connector_name",
    "data_asset_name": "w_customer_d",
    "limit": 2000,
    "batch_spec_passthrough":{
        "schema_name": "masterdata"}
}

this throws this error

_____________________________________________________________________________________________________________
Loaded ExpectationSuite "masterdata.w_customer_d.warning" containing 0 expectations.
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-3af52e39fcdb> in <module>
     39 validator = context.get_validator(
     40     batch_request=BatchRequest(**batch_request),
---> 41     expectation_suite_name=expectation_suite_name,
     42 )
     43 column_names = [f'"{column_name}"' for column_name in validator.columns()]

~\AppData\Roaming\Python\Python37\site-packages\great_expectations\data_context\data_context.py in get_validator(self, datasource_name, data_connector_name, data_asset_name, batch_request, batch_request_list, batch_data, data_connector_query, batch_identifiers, limit, index, custom_filter_function, expectation_suite_name, expectation_suite, create_expectation_suite_with_name, batch_spec_passthrough, sampling_method, sampling_kwargs, splitter_method, splitter_kwargs, runtime_parameters, query, path, batch_filter_parameters, expectation_suite_ge_cloud_id, **kwargs)
   1842                     path=path,
   1843                     batch_filter_parameters=batch_filter_parameters,
-> 1844                     **kwargs,
   1845                 )
   1846             )

~\AppData\Roaming\Python\Python37\site-packages\great_expectations\data_context\data_context.py in get_batch_list(self, datasource_name, data_connector_name, data_asset_name, batch_request, batch_data, data_connector_query, batch_identifiers, limit, index, custom_filter_function, batch_spec_passthrough, sampling_method, sampling_kwargs, splitter_method, splitter_kwargs, runtime_parameters, query, path, batch_filter_parameters, **kwargs)
   1657             # TODO: Raise a warning if any parameters besides batch_requests are specified
   1658             return datasource.get_batch_list_from_batch_request(
-> 1659                 batch_request=batch_request
   1660             )
   1661         elif any([batch_data is not None, query, path, runtime_parameters]):

~\AppData\Roaming\Python\Python37\site-packages\great_expectations\datasource\new_datasource.py in get_batch_list_from_batch_request(self, batch_request)
    188                     batch_markers,
    189                 ) = data_connector.get_batch_data_and_metadata(
--> 190                     batch_definition=batch_definition
    191                 )
    192                 new_batch: Batch = Batch(

~\AppData\Roaming\Python\Python37\site-packages\great_expectations\datasource\data_connector\data_connector.py in get_batch_data_and_metadata(self, batch_definition)
     98         batch_spec: BatchSpec = self.build_batch_spec(batch_definition=batch_definition)
     99         batch_data, batch_markers = self._execution_engine.get_batch_data_and_markers(
--> 100             batch_spec=batch_spec
    101         )
    102         self._execution_engine.load_batch_data(batch_definition.id, batch_data)

~\AppData\Roaming\Python\Python37\site-packages\great_expectations\execution_engine\sqlalchemy_execution_engine.py in get_batch_data_and_markers(self, batch_spec)
   1066             else:
   1067                 selectable: Selectable = self._build_selectable_from_batch_spec(
-> 1068                     batch_spec=batch_spec
   1069                 )
   1070 

~\AppData\Roaming\Python\Python37\site-packages\great_expectations\execution_engine\sqlalchemy_execution_engine.py in _build_selectable_from_batch_spec(self, batch_spec)
   1009             sa.select("*")
   1010             .select_from(
-> 1011                 sa.table(table_name, schema=batch_spec.get("schema_name", None))
   1012             )
   1013             .where(split_clause)

TypeError: table() got an unexpected keyword argument 'schema'

**Comment by talagluck** [comment]
Hi @mpalfi - thanks for reaching out. If you remove the `limit` key from the `batch_request`, do you still have the same issue?

**Comment by mpalfi** [comment]
Thanks for getting back with the suggestions. I tried it now, but the end result with the error remained the same.

**Comment by talagluck** [comment]
Hi @mpalfi - thanks for the update. This has come up elsewhere, and I believe it has to do with the SQL Alchemy version. Are you running a version <1.4? I believe the error should go away if you run >= 1.3.18.

**Comment by talagluck** [comment]
I believe this Issue is resolved by #3938. @mpalfi, I believe your issue will be resolved by using a newer version of SQL Alchemy, and is also a duplicate of #3417, so I'm going to close this for now. Please feel free to open a new issue if that is not the case. Thanks!
